src/calculators/BinaryEligibilityOracleEarningPowerCalculator.sol

```diff
@@ -177,6 +183,33 @@ contract BinaryEligibilityOracleEarningPowerCalculator is Ownable, IEarningPower
     lastOracleUpdateTime = block.timestamp;
   }
 
+  /// @notice Updates the eligibility scores of multiple delegatees in a single transaction.
+  /// @dev This function can only be called by the authorized `scoreOracle` address.
+  /// @dev If the oracle is paused, the update will be reverted.
+  /// @dev If any of the delegatees' scores is locked, the update will be reverted.
+  /// @dev Updates are processed sequentially. If the same delegatee appears multiple times in the
+  /// array, the last update will be applied.
+  /// @dev The `lastOracleUpdateTime` is only updated if the array contains at least one update.
+  /// @param _delegateeScoreUpdates An array of DelegateeScoreUpdate structs containing delegatee
+  /// addresses and their new scores.
+  function updateDelegateeScores(DelegateeScoreUpdate[] calldata _delegateeScoreUpdates) public {
+    if (msg.sender != scoreOracle) {
+      revert BinaryEligibilityOracleEarningPowerCalculator__Unauthorized("not oracle", msg.sender);
+    }
+    if (isOraclePaused) {
+      revert BinaryEligibilityOracleEarningPowerCalculator__DisallowedWhilePaused();
+    }
+    for (uint256 _i = 0; _i < _delegateeScoreUpdates.length; _i++) {
+      address _delegatee = _delegateeScoreUpdates[_i].delegatee;
+      uint256 _newScore = _delegateeScoreUpdates[_i].newScore;
+      if (delegateeScoreLockStatus[_delegatee]) {
```

Comment:
This check can be moved to the top of the `_updateDelegateeScore` method so it will also be shared